test/e2e/e2e.go

```diff
@@ -1357,187 +1388,216 @@ var _ = ginkgo.Describe("e2e multiple ecmp external gateway validation", func()
-			framework.Failf("failed to start external gateway test container %s: %v", gwContainer1, err)
-		}
-		// retrieve the container ip of the external gateway container
-		exGWIp1, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, gwContainer1)
-		if err != nil {
-			framework.Failf("failed to start external gateway test container: %v", err)
-		}
-		// trim newline from the inspect output
-		exGWIp1 = strings.TrimSuffix(exGWIp1, "\n")
-		if ip := net.ParseIP(exGWIp1); ip == nil {
-			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainer1, exGWIp1)
+		if addressesv6.srcPodIP != "" && addressesv6.nodeIP != "" {
+			testIPV6 = true
 		}
 
-		// start the second container that will act as an external gateway
-		_, err = runCommand("docker", "run", "-itd", "--privileged", "--network", ciNetworkName, "--name", gwContainer2, "centos/tools")
-		if err != nil {
-			framework.Failf("failed to start external gateway test container %s: %v", gwContainer2, err)
-		}
-		// retrieve the container ip of the external gateway container
-		exGWIp2, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, gwContainer2)
-		if err != nil {
-			framework.Failf("failed to start external gateway test container: %v", err)
+		// This sets up a listener that replies with the hostname, both on tcp and on udp
+		setupListeners := func(container, address string) {
+			cmd := []string{"docker", "exec", container, "bash", "-c", fmt.Sprintf("while true; do echo $(hostname) | nc -l -u %s %d | exit; done &", address, externalUDPPort)}
+			_, err = runCommand(cmd...)
+			framework.ExpectNoError(err, "failed to setup listener on ", address, container)
+
+			cmd = []string{"docker", "exec", container, "bash", "-c", fmt.Sprintf("while true; do echo $(hostname) | nc -l %s %d | exit; done &", address, externalTCPPort)}
+			_, err = runCommand(cmd...)
+			framework.ExpectNoError(err, "failed to setup listener on ", address, container)
 		}
-		// trim newline from the inspect output
-		exGWIp2 = strings.TrimSuffix(exGWIp2, "\n")
-		if ip := net.ParseIP(exGWIp2); ip == nil {
-			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainer2, exGWIp2)
+
+		// The target ips are addresses added to the lo of each container.
+		// By setting the gateway annotation and using them as destination, we verify that
+		// the routing is able to reach the containers.
+		// A route back to the src pod must be set in order for the ping reply to work.
+		for _, containerName := range []string{gwContainer1, gwContainer2} {
+
+			ginkgo.By(fmt.Sprintf("Setting up the destination ips to %s", containerName))
+			for _, address := range addressesv4.targetIPs {
+				_, err = runCommand("docker", "exec", containerName, "ip", "address", "add", address+"/32", "dev", "lo")
+				framework.ExpectNoError(err, "failed to add the loopback ip to dev lo on the test container %s", containerName)
```

Comment:
Might be useful to print out the address that failed to be added for future debugging